a4_APIweatherApp.py

```diff
@@ -0,0 +1,148 @@
+      "2024-08-02T22:00",
+      "2024-08-02T23:00",
+      "2024-08-03T00:00",
+      "2024-08-03T01:00"]
+        temps = [19.4, 18.8, 18.2, 17.8, 17.7, 17.7, 17.9, 18, 18.1, 18.6, 18.8, 18.8, 18.9, 19.2, 19.4, 19.6, 19.4, 19.5, 19.4, 19, 18.5, 18, 17.6, 17.3]
+        mockArray2d = [dates, temps, 0]
+        return mockArray2d
+
+
+class APIService(Service):
+    def get_weather_data(self):
+        '''
+        This method asks for user input of the location of where they would
+        like to get temperature data from and then makes an API request call for weather and time in that location.
+        It puts the data into their respective arrays.
+        Finally, puts the data into a 2D array with a marker to signify
+        that it is an API data set.
+        '''
+        '''
+        base_url = "https://api.open-meteo.com/v1/forecast"  
+        params = {}  
+        params['latitude'] = latitude    # float  
+        params['longitude'] = longitude    # float  
+        params['hourly'] = daily_weather_variable    # string with "temperature_2m"  
+
+        response = requests.get(base_url, params=params)
+        '''
+        latitude = input("Please write your latitude: ")
+        longitude = input("Please write your longitude: ")
+        response = requests.get(f'https://api.open-meteo.com/v1/forecast?{latitude}=52.52&{longitude}=13.41&hourly=temperature_2m')
+        if response.status_code == 200 and 'hourly' in response and 'time' in response and 'temperature_2m' in response:
+            dates = response.json()['hourly']['time']
+            temps = response.json()['hourly']['temperature_2m']
+        apiArray2d = [dates, temps, 1]
+        return apiArray2d
+        
+
+
+class ServiceFactory(ABC):
+    def create_service(self, num):
+        '''
+        This method generates either an API or Mock Service instance based on
+        the input value.
+        '''
+        if num == Environment.PROD:  
+            return APIService()  
+        elif num == Environment.TEST:  
+            return MockService()
+
+
+class Handler:
+    def __init__(self, service): #the service will be injected
+        self.sv = service
+
+    def plot(self):
+        '''
+        This method takes the data of the given Service and generates a plot
+        of the temperature over time.
+        '''
+
+        values = []
+        values = self.sv.get_weather_data() #error is here; Attribute of service
+            #returns the 2d array
+        plt.scatter(values[0], values[1])
+        if values[2] == 1:
+            plt.title("API Service")
+        elif values[2] == 0:
+            plt.title("Mock Service")
+        plt.show()
+
+
+def main():
+    '''
+    This main function calls both the Mock and API Services and makes sure
+    that it outputs a plot of the temperature over time for both.
+    '''
+    #API
+    api_factory = ServiceFactory()
+    api = api_factory.create_service(1) #api is the instance of the API Service
```

Comment:
I think you just need to pass `Environment.PROD` instead of `1` directly for the conditional in `create_service` to return a valid value. 

If you are getting a None here, that means your none of your if statements are returning true.